### PR TITLE
Jlisowskyy/setup kernel site test framework

### PR DESCRIPTION
**Pull request by Jlisowskyy**
Things done in this PR:
- Added test framework on kernel site
- Fully working test fixtures, tests, fail tests 
- Automatic test adding to lists etc
- Test framework fully working
- Added tests of test framework
- Added support for manual tests
- Added traces inside host site code
- Rewritten old tests
- Removed old test functionality

### CODE REVIEW

File: alkos/kernel/CMakeLists.txt [hunk 1]
```diff
@@ -14,10 +14,11 @@ include_directories(include)
 include_directories(${SYSROOT}/usr/include)
 include_directories(test)
 
-add_compile_definitions(ALKOS_TEST)
-
 add_compile_definitions(__ALKOS_KERNEL__=1)
 
+# Uncomment to start test framework
```
Comment (kryczkal) [inline alkos/kernel/CMakeLists.txt:19]: ...

File: alkos/kernel/CMakeLists.txt [hunk 2]
```diff
@@ -14,10 +14,11 @@ include_directories(include)
...
+# add_compile_definitions(__ALKOS_TEST__=1)
+
 # Defining this macro enables debug terminal
 add_compile_definitions(__DEBUG_TERMINAL_TEST__=1)
```
Comment (kryczkal) [inline alkos/kernel/CMakeLists.txt:23]: should not be called test then

File: alkos/kernel/arch/x86_64/drivers/serial_port_qemu/serial_qemu.cpp [hunk 3]
```diff
@@ -54,7 +54,8 @@ void QemuTerminalInit()
      * 0xC=0b1100, 0x7=0b0111, 0xC7=0b1100'0111
      *
      * This flags states that:
-     * - set up that after received 14 bytes in the FIFO interrupt will be triggered (bits [6-7] set to 11)
+     * - set up that after received 14 bytes in the FIFO interrupt will be triggered (bits [6-7] set
```
Comment (kryczkal) [inline alkos/kernel/arch/x86_64/drivers/serial_port_qemu/serial_qemu.cpp:57]: make this understandable please

**Review by kryczkal — APPROVED** [review]

File: alkos/kernel/CMakeLists.txt (hunk 1 above)
Comment (Jlisowskyy) [reply alkos/kernel/CMakeLists.txt:19]: Will be fixed in cicd pr. Cmake files are not ready yet

File: alkos/kernel/CMakeLists.txt (hunk 2 above)
Comment (Jlisowskyy) [reply alkos/kernel/CMakeLists.txt:23]: Will be done in cmake pr

File: alkos/kernel/include/types.hpp [hunk 4]
```diff
@@ -22,4 +22,21 @@ using i64 = int64_t;
 using f32 = float;
 using f64 = double;
 
+/* simplified unsigned int types */
```
[the remark this answers is not in the document]
Comment (Jlisowskyy) [reply alkos/kernel/include/types.hpp:25]: fixed

**Pull request merged**